pkg/apis/extensions/types.go

```diff
@@ -437,6 +449,22 @@ type RollingUpdateDaemonSet struct {
 	MaxUnavailable intstr.IntOrString
 }
 
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
```

Comment:
There were some discussion about that, but sig-apps didn't want to break current API and behavior, hence this new strategy. Keeping things as stable as possible...